api/projects/projects.js

```diff
@@ -0,0 +1,193 @@
+    name === undefined ||
+    description === undefined
+  ) {
+    console.log(req.body, 54);
+    res.status(400).send();
+    return;
+  }
+  console.log(founder, name, description, 29);
+  let newProject = await db.Project.create({
+    name: name,
+    description: description,
+    projectOwner: founder,
+    members: [],
+    applicants: [],
+    signatures: [],
+    createdAt: Date.now(),
+  });
+  newProject.members.push(founder);
+  newProject.save();
+  const success = "project created successfully by " + founder;
+  res.status(200).send(success);
+});
+
+// #13 Edit Project
+router.put("/project", (req, res) => {
+  let externalId = req.body.externalId;
+  let description = req.body.description;
+  let remainingMembers = req.body.remainingMembers;
+  let remainingApplicants = req.body.remainingApplicants;
+  // todo: should probably rewrite as, "membersToRemove" "applicantsToRemove" etc
+  console.log(req.body, 88);
+  db.Project.findOneAndUpdate(
+    { externalId: externalId },
+    {
+      description: description,
+      members: remainingMembers,
+      applicants: remainingApplicants,
+    },
+    {
+      new: true,
+    }
+  )
+    .then((newDoc) => {
+      res.status(200).json(newDoc);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+});
+
+router.put("/project/:id", (req, res) => {
+  // attach so and so's name to the arr of ppl who voted for it
+  let projectId = req.query.id;
+  const voter = req.body.voter;
+  // TODO: finish
+});
+
+router.put("/project/add", async (req, res) => {
+  // so someone can add an applicant to a project
+  let externalId = req.body.externalId;
+  let applicantToAdd = req.body.newApplicant;
+  console.log(135, externalId);
+  let projectToUpdate = await db.Project.findOne({ externalId: externalId });
+  projectToUpdate.applicants.push(applicantToAdd);
+  projectToUpdate.save();
+  res.status(200).send("successful update");
+  // TODO: test either via postman or actual tests
+});
+
+router.put("/update/convert", async (req, res) => {
+  // so someone who was previously an application can be converted to a member status
+  let projectId = req.body.id;
+  let promotionTarget = req.body.promotionTarget;
+  let projectToUpdate = await db.Project.updateOne(
+    // fixme: this is untested & I smell a bug here
+    { _id: projectId },
+    { $pullAll: { applicants: [promotionTarget] } }
+  );
+  projectToUpdate.members.push(promotionTarget);
+  projectToUpdate.save();
+  res.status(200).send("successful update");
+});
+
+router.delete("/deleteProject/:projectId", (req, res) => {
+  // hypothetical route
+  let authorizedUser = req.body.authorizedUser;
+  let signedAuth = req.body.signedAuth;
+  db.Project.findOneAndDelete(
+    {
+      projectOwner: authorizedUser,
+      authLine: signedAuth,
+    },
+    function (err, deleted) {
+      if (err) {
+        res
+          .status(400)
+          .send("failed to delete -- are you authorized to do this?");
```

Comment:
Can the front end know this by the http status code? Why do you need the failure message?